### PR TITLE
Add partylist event

### PR DESCRIPTION
**Pull request by nonowazu**
**Purpose**
You know what I'm here for. I'm here to make friends and implement PartyList.
...and I'm all out of Friends.

**New Features**
Adds PartyList event and helping parser for act

**Bug Fixes**
N/A

**Before/After**
Before: No PartyList event
After: PartyList event

**Additional Context**
I listened to the Smash Mouth cover of *Why Can't We Be Friends* while I wrote this. Now I have no friends.

### CODE REVIEW

File: nari/io/reader/actlogutils/party.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+""""Parses partylist events from act log line"""
+from datetime import datetime, time
```
Comment (lunarrize) [inline nari/io/reader/actlogutils/party.py:2]: You're not using `time`

File: nari/io/reader/actlogutils/party.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    # Param layout from ACT
+    # 0 - number of actor IDs
+    # 1 - 0 or more actor IDs for the party. First entry is always your actor;
+    #     subsequent actors are those of your party (up to the amount listed 
+    #     in param 0). Any further actors are actors that are a part of your
+    #     alliance
+    amount = int(params[0])
+    if amount == 0:
+        return # TODO: perhaps send a 'PartyDisband' event?
+    else:
```
Comment (lunarrize) [inline nari/io/reader/actlogutils/party.py:20]: You don't need this else the `if` already returns

File: nari/io/reader/actlogutils/party.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        # amount is > 1; grab local party first
+        ids = [int(i) for i in params[1:amount+1]]
+        # TODO: parse other_ids
```
Comment (lunarrize) [inline nari/io/reader/actlogutils/party.py:23]: Go ahead and implement this

File: nari/types/event/party.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+from nari.types.event import Event
+
+
+class PartyList(Event): # pylint: disable=too-few-public-methods
+    """Represents changes to the party list"""
+    def __init__(self, *, # pylint: disable=dangerous-default-value
+                 timestamp: datetime,
+                 ids: List[int],
+                 other_ids: List[int] = [],
+    ):
```
Comment (lunarrize) [inline nari/types/event/party.py:14]: while this is okay I think other modules visually align this with the opening `(`

**Review by lunarrize — CHANGES_REQUESTED** [review]
needs some changes but otherwise fine

**Comment by nonowazu** [comment]
Changes in – sorry for the delay!

File: nari/io/reader/actlogutils/party.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    #     subsequent actors are those of your party (up to the amount listed 
+    #     in param 0). Any further actors are actors that are a part of your
+    #     alliance
+    amount = int(params[0])
+    if amount == 0:
+        return # TODO: perhaps send a 'PartyDisband' event?
+
+    # amount is > 1; grab local party first
+    ids = [int(i) for i in params[1:amount+1]]
+    other_ids = [int(i) for i in params[amount+1:]]
```
Comment (lunarrize) [inline nari/io/reader/actlogutils/party.py:23]: both of these should be `int(i, 16)`

**Review by lunarrize — CHANGES_REQUESTED** [review]

**Pull request merged**